lib/node_modules/@stdlib/repl/lib/completer_engine.js

```diff
@@ -0,0 +1,747 @@
+* @memberof CompleterEngine.prototype
+* @type {Function}
+* @returns {void}
+*/
+setNonEnumerableReadOnly( CompleterEngine.prototype, '_navigateDown', function navigateDown() {
+	var columns = this._completionColumns();
+	var rows = this._completionRows( columns );
+
+	// Move to the next row...
+	if ( this._idx === -1 ) {
+		this._idx = 0;
+		this._updateCompletions();
+	} else if ( this._idx + columns < rows * columns ) {
+		this._idx += columns;
+		this._updateCompletions();
+	}
+});
+
+/**
+* Navigate to the left in the completions grid.
+*
+* @private
+* @name _navigateLeft
+* @memberof CompleterEngine.prototype
+* @type {Function}
+* @param {string} data - input data
+* @param {Object} key - key object
+* @returns {void}
+*/
+setNonEnumerableReadOnly( CompleterEngine.prototype, '_navigateLeft', function navigateLeft( data, key ) {
+	// If on current line, trigger default behavior and stop navigating...
+	if ( this._idx === -1 ) {
+		this._ttyWrite.call( this._rli, data, key );
+		return;
+	}
+	// If navigating, move back an index...
+	if ( this._idx > 0 ) {
+		this._idx -= 1;
+		this._updateCompletions();
+	}
+});
+
+/**
+* Navigate to the right in the completions grid.
+*
+* @private
+* @name _navigateRight
+* @memberof CompleterEngine.prototype
+* @type {Function}
+* @param {string} data - input data
+* @param {Object} key - key object
+* @returns {void}
+*/
+setNonEnumerableReadOnly( CompleterEngine.prototype, '_navigateRight', function navigateRight( data, key ) {
+	var columns = this._completionColumns();
+	var rows = this._completionRows( columns );
+
+	// If on current line, trigger default behavior and stop navigating...
+	if ( this._idx === -1 ) {
+		this._ttyWrite.call( this._rli, data, key );
+		return;
+	}
+	// If navigating, move ahead an index...
+	if ( this._idx + 1 < min( rows * columns, this._completionsList.length ) ) {
+		this._idx += 1;
+		this._updateCompletions();
+	}
+});
+
+/**
+* Checks whether the user is currently navigating completions.
+*
+* @name isNavigating
+* @memberof CompleterEngine.prototype
+* @type {Function}
+* @returns {boolean} boolean indicating whether user is currently navigating completions
+*/
+setNonEnumerableReadOnly( CompleterEngine.prototype, 'isNavigating', function isNavigating() {
+	return this._isNavigating;
+});
+
+/**
+* Closes the completer engine.
+*
+* @private
+* @name closeCompleter
+* @memberof CompleterEngine.prototype
+* @type {Function}
+* @returns {void}
+*/
+setNonEnumerableReadOnly( CompleterEngine.prototype, 'closeCompleter', function closeCompleter() {
+	// Check if completer is already closed...
+	if ( !this._isNavigating ) {
+		return;
+	}
+	// Reset completer parameters:
+	this._isNavigating = false;
+	this._idx = -1;
+
+	// Clear completions output:
+	readline.clearScreenDown( this._ostream );
+
+	// NOTE: `clearScreenDown` seems to behave abnormally in this case by also clearing the `ostream` to the right of the cursor. Hence writing it again below:
```

Comment:
Is this abnormal behavior observed across terminals/platforms? Do we need to include a workaround in case the behavior is considered a bug and later changed?